src/utils/sovryn/sovryn-network.ts

```diff
@@ -22,13 +22,19 @@ export class SovrynNetwork {
   public databaseContractList: Contract[] = [];
 
   constructor() {
-    this.initReadWeb3(currentChainId)
+    this.init();
+  }
+
+  private init(retry = 0) {
+    this.initReadWeb3(currentChainId, retry)
       .then(async () => {
         const gasPrice = await this.getOnChainGasPrice();
         gas.set(gasPrice);
         this.refreshGasPrice();
       })
-      .catch();
+      .catch(() => {
+        this.init(++retry);
+      });
```

Comment:
This was more or less intentional. My feature only deals with a connection problem to the rpc. initReadWeb3 does not verify the connection, so I did not see the need to change that.

If we catch an error from within initReadWeb3, changing the rpc will not help, because the problem has to be somewhere else.

On the other hand, it does not hurt, so I will take the suggestion.

Reply:
Yeah I took a second look you're right, the init doesn't fail if the rpc is unavailable, propagating potential errors, shouldn't hurt though.